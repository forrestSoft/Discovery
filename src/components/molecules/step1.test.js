import React from 'react'
import {shallow, mount} from 'enzyme'
import shallowToJson from 'enzyme-to-json'

import Step1 from './step1'

it('render matches snapshot', () => {
  const mock = jest.fn().mockReturnValue('onClick')
  const step1 = shallowToJson(shallow(
  	<Step1 
      handleChange={mock}
      data={
        {
          email: '',
          emailValid: false,
          checkboxChecked: false
        }
      }
    />
  )) 
  expect(step1).toMatchSnapshot();
});

it('render with data matches snapshot', () => {
  const mock = jest.fn().mockReturnValue('onClick')
  const step1 = shallowToJson(shallow(
    <Step1 
      handleChange={mock}
      data={
        {
          email: 'bob',
          emailValid: false,
          checkboxChecked: true
        }
      }
    />
  )) 
  expect(step1).toMatchSnapshot();
});

it('render with data disables button', () => {
  const mock = jest.fn().mockReturnValue(jest.fn())
  const mock2 = jest.fn().mockReturnValue(jest.fn())

  const step1 = mount(
    <Step1 
      handleChange={mock}
      handleClick={mock2}
      data={
        {
          email: 'bob',
          emailValid: false,
          checkboxChecked: true
        }
      }
    />
  )
  
  expect(step1.find('button').is('[disabled]')).toBe(true)
});

it('ensures passed handlers are called', () => {
  const mock = jest.fn().mockReturnValue(jest.fn())
  const mock2 = jest.fn().mockReturnValue(jest.fn())
  
  const step1 = mount(
    <Step1 
      handleChange={mock}
      handleClick={mock2}
      data={
        {
          email: 'bob',
          emailValid: true,
          checkboxChecked: true
        }
      }
    />
  )
  
  step1.find('input').first().simulate('change')
  expect(mock).toBeCalled()

  step1.find('input').at(1).simulate('change')
  expect(mock).toBeCalled()

  step1.find('button').simulate('click')
  expect(mock2).toBeCalled()
});

it('ensures passed handlers are not called', () => {
  const mock = jest.fn().mockReturnValue(jest.fn())
  const mock2 = jest.fn().mockReturnValue(jest.fn())
  
  const step1 = mount(
    <Step1 
      handleChange={mock}
      handleClick={mock2}
      data={
        {
          email: 'bob',
          emailValid: false,
          checkboxChecked: true
        }
      }
    />
  )
  
  step1.find('button').simulate('click')
  expect(mock2).not.toBeCalled()
});
