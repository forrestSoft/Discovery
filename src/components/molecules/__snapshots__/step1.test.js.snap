// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches snapshot 1`] = `
<Fragment>
  <div
    className="App step1"
  >
    <Header
      text="join the list"
    />
    <div>
      <SubText
        text="i am subText"
      >
        <Text
          type="span"
        >
          Sign up for
        </Text>
        <Text
          type="span"
        >
          the TLC newsletter.
        </Text>
      </SubText>
      <div
        className="inputWrapper"
      >
        <TextInput
          onChange="onClick"
          placeholder="email address"
          valid={false}
          value=""
        />
        <Button
          disabled={true}
          text="Next"
        />
      </div>
      <div
        className="helperTextWrapper"
      >
        <CheckBox
          checked={false}
          onChange="onClick"
          valid={false}
        />
        <Text
          type="span"
        >
          I agree to receive information from Discovery Communications in accordance with the following
          <Link
            text="Privacy Policy"
            url="https://google.com"
          />
        </Text>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`render with data matches snapshot 1`] = `
<Fragment>
  <div
    className="App step1"
  >
    <Header
      text="join the list"
    />
    <div>
      <SubText
        text="i am subText"
      >
        <Text
          type="span"
        >
          Sign up for
        </Text>
        <Text
          type="span"
        >
          the TLC newsletter.
        </Text>
      </SubText>
      <div
        className="inputWrapper"
      >
        <TextInput
          onChange="onClick"
          placeholder="email address"
          valid={false}
          value="bob"
        />
        <Button
          disabled={true}
          text="Next"
        />
      </div>
      <div
        className="helperTextWrapper"
      >
        <CheckBox
          checked={true}
          onChange="onClick"
          valid={true}
        />
        <Text
          type="span"
        >
          I agree to receive information from Discovery Communications in accordance with the following
          <Link
            text="Privacy Policy"
            url="https://google.com"
          />
        </Text>
      </div>
    </div>
  </div>
</Fragment>
`;
